haystack/nodes/prompt/prompt_node.py

```diff
@@ -562,6 +504,9 @@ def __init__(
         devices: Optional[List[Union[str, torch.device]]] = None,
     ):
         super().__init__()
+        self.prompt_templates: Dict[str, PromptTemplate] = {
+            pt.name: pt for pt in PromptNode.get_predefined_prompt_templates()  # type: ignore
```

Comment:
I checked why we need the  `#  type: ignore` here. We get an `error: Key expression in dictionary comprehension has incompatible type "Optional[str]"; expected type "str"` because mypy thinks `pt.name` might be `None`. I agree that we can ignore that.